Data/PackageData.cs

```diff
@@ -6,15 +6,91 @@
 using System.Net.Http;
 using System.IO.Compression;
 using System.Xml.Linq;
-using Mono.Cecil;
 using System.Threading;
 using System.Diagnostics;
 using System.Text.RegularExpressions;
+using System.Text;
 
 namespace FuGetGallery
 {
+    public class Entry
+    {
+        public PackageData Package { get; set; }
+        public string FullName { get; set; }
+        public int Mode { get; set; }
+        public int CompressedSize { get; set; }
+        public int ExpandedSize { get; set; }
+        public int EntryOffset { get; set; }
+        public int Length => ExpandedSize;
+        public override string ToString ()
+        {
+            return $"{FullName} {Mode}";
+        }
+
+        public Stream Open ()
+        {
+            return OpenAsync ().GetAwaiter ().GetResult ();
+        }
+
+        public async Task<Stream> OpenAsync () {
+
+            Debug.WriteLine ("Downloading: " + this.FullName + " " + this.CompressedSize);
+            var url = this.Package.DownloadUrl;
+            var client = this.Package.Client;
+
+            var req = new HttpRequestMessage () {
+                Method = HttpMethod.Get,
+                RequestUri = new Uri (url, UriKind.Absolute),
+                Version = new Version (1, 1),
+            };
+            req.Headers.Add ("Range", "bytes=" + EntryOffset + "-" + (EntryOffset + 30 -1));
+
+            var resp = await client.SendAsync (req);
+            var buf = await resp.Content.ReadAsByteArrayAsync ();
+
+            var fileNameLen = buf.GetInt16 (26);
+            var fileExtraLen = buf.GetInt16 (28);
+
+            var ms = new MemoryStream (this.ExpandedSize);
+
+
+            req = new HttpRequestMessage () {
+                Method = HttpMethod.Get,
+                RequestUri = new Uri (url, UriKind.Absolute),
+                Version = new Version (1, 1),
+            };
+
+            var deflateStart = EntryOffset + 30 + fileNameLen + fileExtraLen;
+
+
+            req.Headers.Add ("Range", "bytes=" + deflateStart + "-" + (deflateStart + CompressedSize - 1));
+
+            resp = await client.SendAsync (req);
+            var stream = await resp.Content.ReadAsStreamAsync();
+
+            var ds = new DeflateStream (stream, CompressionMode.Decompress, false);
+            ds.CopyTo (ms);
+            ms.Position = 0;
+            return ms;
+        }
+    }
+
+    static class BufferHelpers
+    {
+        public static int GetInt32 (this byte[] buffer, int offset)
+        {
+            return BitConverter.ToInt32 (buffer.AsSpan().Slice(offset, 4));
```

Comment:
```suggestion
            return BitConverter.ToInt32 (buffer, offset);
```

seems easier to understand (to me), and is likely to be just as efficient